crates/rattler_conda_types/src/package/entry_point.rs

```diff
@@ -0,0 +1,97 @@
+//! A struct for a single Python entry point.
+
+use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
+use std::fmt::Display;
+use std::str::FromStr;
+
+/// A struct for a single Python entry point. An entry point is a command that
+/// runs a function in a Python module. For example, the entry point
+/// `jlpm = jupyterlab.jlpmapp:main` will run the `main` function in the
+/// `jupyterlab.jlpmapp` module when the command `jlpm` is run on the command line.
+/// The main usage for entry points is in `noarch: python` packages.
+///
+/// The entry point is represented as a string in the format
+/// `<command> = <module>:<function>`. The command is the name of the command that
+/// will be run on the command line. The module is the name of the Python module
+/// that contains the function. The function is the name of the function to run.
+///
+/// The entry point is parsed from a string using the [`FromStr`] trait. The
+/// [`Display`] trait is implemented for the entry point to convert it back to a
+/// string.
+#[derive(Debug, Clone)]
+pub struct EntryPoint {
+    /// The name of the command that will be available on the command line.
+    pub command: String,
+
+    /// The name of the Python module that contains the function.
+    pub module: String,
+
+    /// The name of the function to run.
+    pub function: String,
+}
+
+impl FromStr for EntryPoint {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let mut parts = s.splitn(2, '=');
```

Comment:
You can use `split_once` instead of `splitn`. That will give you a tuple of the left and right side.

Reply:
Cool!, done